jsonnet/components/control-plane.libsonnet

```diff
@@ -141,6 +141,42 @@ function(params)
       },
     },
 
+    // This adds a kubelet ServiceMonitor for special use with
+    // prometheus-adapter
+    serviceMonitorKubeletResourceMetrics: self.serviceMonitorKubelet {
+      metadata+: {
+        name: 'kubelet-resource-metrics',
+      },
+      spec+: {
+        endpoints:
+          std.filterMap(
+            function(e)
+              'path' in e && e.path == '/metrics/cadvisor'
+            ,
+            function(e)
+              e {
+                path: '/metrics/resource',
+                honorTimestamps: true,
+                metricRelabelings: [
+                  {
+                    sourceLabels: ['__name__'],
+                    action: 'keep',
+                    regex: 'container_cpu_usage_seconds_total|container_memory_working_set_bytes',
```

Comment:
Hmm after thinking about this a little more: isn't this the same `scrape_error` metric we already get from the control-plane ServiceMonitor? I.e. it will just duplicate the "Failed to scrape container metrics" signal, since it scrapes the same kubelet instance?

Reply:
I don't think so, my understanding is that kubelet will return `scrape_error 1` if it fails to gather container metrics for some reason while it will still return 200 OK to Prometheus meaning `up 1`.

Reply:
sorry I misread your comment but I think it's still useful to collect scrape_error for both service monitors. Also it's one series per kubelet so not a cardinality concern.

Reply:
I suppose I'm not quite clear what this metric means :thinking: 
I was under the impression this signals that kubelet is unable to get container metrics, so this would just add the same information since its the same kubelet we are scraping.
But its actually related to the service monitor?

Reply:
no it's indeed related to the kubelet collecting container metrics but I wouldn't necessarily assume that both endpoints /metrics/cadvisor and /metrics/resource return the same value (it will probably but who knows :shrug:).